src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -0,0 +1,599 @@
+      callback.onCompletion(null, Errors.consumerFormatMismatch());
+      return null;
+    }
+
+    int workerId = nextWorker.getAndIncrement() % workers.size();
+    KafkaConsumerWorker worker = workers.get(workerId);
+    return worker.readRecords(
+        state, timeout, maxBytes,
+        new ConsumerWorkerReadCallback<ClientK, ClientV>() {
+          @Override
+          public void onCompletion(
+              List<? extends ConsumerRecord<ClientK, ClientV>> records, Exception e) {
+            updateExpiration(state);
+            if (e != null) {
+              // Ensure caught exceptions are converted to RestExceptions so the user gets a
+              // nice error message. Currently we don't define any more specific errors because
+              // the old consumer interface doesn't classify the errors well like the new
+              // consumer does. When the new consumer is available we may be able to update this
+              // to provide better feedback to the user.
+              Exception responseException = e;
+              if (!(e instanceof RestException)) {
+                responseException = Errors.kafkaErrorException(e);
+              }
+              callback.onCompletion(null, responseException);
+            } else {
+              callback.onCompletion(records, null);
+            }
+          }
+        }
+    );
+  }
+    
+  public interface CommitCallback {
+
+    public void onCompletion(List<TopicPartitionOffset> offsets, Exception e);
+  }
+
+  public Future commitOffsets(String group, String instance, final String async, final ConsumerOffsetCommitRequest offsetCommitRequest, final CommitCallback callback) {
+    final KafkaConsumerState state;
+    try {
+      state = getConsumerInstance(group, instance);
+    } catch (RestNotFoundException e) {
+      callback.onCompletion(null, e);
+      return null;
+    }
+
+    return executor.submit(new Runnable() {
+      @Override
+      public void run() {
+        try {
+	    List<TopicPartitionOffset> offsets = state.commitOffsets(async, offsetCommitRequest);
+          callback.onCompletion(offsets, null);
+        } catch (Exception e) {
+          log.error("Failed to commit offsets for consumer " + state.getId().toString(), e);
+          Exception responseException = e;
+          if (!(e instanceof RestException)) {
+            responseException = Errors.kafkaErrorException(e);
+          }
+          callback.onCompletion(null, responseException);
+        } finally {
+          updateExpiration(state);
+        }
+      }
+    });
+  }
+
+  public ConsumerCommittedResponse committed(String group, String instance, ConsumerCommittedRequest request) {
+    log.debug("Committed offsets for consumer " + instance + " in group " + group);
+    ConsumerCommittedResponse response = new ConsumerCommittedResponse();	
+    KafkaConsumerState state = getConsumerInstance(group, instance);
+    if (state != null) {
```

Comment:
I think you can skip this check since it looks like if it is missing `getConsumerInstance` will throw an exception.